src/DaprExtension/Bindings/DaprSecretAttribute.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.Dapr
+{
+    using System;
+    using Microsoft.Azure.WebJobs.Description;
+
+    /// <summary>
+    /// Parameter attribute for the Dapr secret input binding.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter)]
+    [Binding]
+    public sealed class DaprSecretAttribute : DaprBaseAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DaprSecretAttribute"/> class.
+        /// </summary>
+        /// <param name="secretStoreName">The name of the secret store to get the secret from.</param>
+        /// <param name="key">The key identifying the name of the secret to get.</param>
+        public DaprSecretAttribute(string secretStoreName, string key)
+        {
+            this.SecretStoreName = secretStoreName ?? throw new ArgumentNullException(nameof(secretStoreName));
+            this.Key = key ?? throw new ArgumentNullException(nameof(key));
+        }
+
+        /// <summary>
+        /// Gets the name of the secret store to get the secret from.
+        /// </summary>
+        [AutoResolve]
+        public string SecretStoreName { get; private set; }
+
+        /// <summary>
+        /// Gets the key identifying the name of the secret to get.
+        /// </summary>
+        [AutoResolve]
+        public string? Key { get; private set; }
```

Comment:
Can't this be non-nullable? It is always set in the constructor and the set method is private.